api/smart_cam/views.py

```diff
@@ -16,31 +19,66 @@
 
 
 class StreamAPI(APIView):
-    def get(self, request):
-        return HttpResponse("Get stream")
+
+    def get(self, request, stream_id=None):
+        if stream_id is None:
+            responseData = Stream.objects.all() 
+            serializer = StreamSerializer(responseData,many=True)
+        else:
+            responseData = Stream.objects.get(id=stream_id)    
+            serializer = StreamSerializer(responseData)
+        return JsonResponse(serializer.data, safe=False)
 
     def post(self, request):
 
+        #Create a new Stream
         feed_url = request.data["url"]
         stream = Stream(url=feed_url, enabled=True)
         stream.save()
 
-        return JsonResponse({"id": stream.id, "url": stream.url})
+        streamIdAndUrl = {"id": stream.id, "url": stream.url}
+        return JsonResponse(streamIdAndUrl,status=status.HTTP_201_CREATED)
 
-    def put(self, request):
+    #safe_get to avoid Stream.DoesNotExist exception
+    def safe_get(self,id):
+        try:
+            got = Stream.objects.get(id=id)
+        except Stream.DoesNotExist:
+            got = None 
+        return got       
 
-        feed_url = request.data["url"]
-        stream = Stream.objects.get(url=feed_url)
-        stream.enabled = request.data["enabled"]
-        stream.save()
+    def put(self, request, stream_id):
+
+        #Update the enabled status of a stream
+        stream = self.safe_get(id=stream_id)
+        updatedEnabledVal = json.loads(request.data["enabled"].lower())
+
+        if stream is not None:
```

Comment:
variables should be name like python's standard guidelines (not JS)

`updatedEnabledVal` => `updated_enabled_val`
...
...